cpp/src/strings/repeat_strings.cu

```diff
@@ -134,10 +134,36 @@ struct compute_size_and_repeat_fn {
   }
 };
 
+/**
+ * @brief Check if the size of the output strings column exceeds the maximum indexable value.
+ */
+void check_overflow(strings_column_view const& input,
+                    size_type repeat_times,
+                    rmm::cuda_stream_view stream)
+{
+  size_type size_start, size_end;
+  CUDA_TRY(cudaMemcpyAsync(&size_start,
+                           input.offsets().template data<size_type>() + input.offset(),
+                           sizeof(size_type),
+                           cudaMemcpyDeviceToHost,
+                           stream.value()));
+  CUDA_TRY(
+    cudaMemcpyAsync(&size_end,
+                    input.offsets().template data<size_type>() + input.offset() + input.size(),
+                    sizeof(size_type),
+                    cudaMemcpyDeviceToHost,
+                    stream.value()));
+  stream.synchronize();
+
+  CUDF_EXPECTS(size_end - size_start <= std::numeric_limits<size_type>::max() / repeat_times,
+               "The output strings have total size that exceeds the maximum allowed size.");
+}
```

Comment:
This approach does not appear very useful. It only tells you the output is too big. It does not really help the caller solve the problem. Perhaps the caller would just simply divide/slice the input in half and try again? If so, then this seem acceptable for this API.

The design we had previously proposed to handle this will to build a separate API that returns the output sizes (as an int32 column). The caller can then use that to determine how to partition (i.e. slice) the input columns and make separate calls for each partition.

Reply:
Sorry that I misunderstood your idea. I'm reworking this now.